src/mito2/src/memtable/bulk/part.rs

```diff
@@ -0,0 +1,702 @@
+        else {
+            return Ok(false);
+        };
+
+        let arrow_schema = arrow_record_batch.schema();
+        {
+            let mut writer = ArrowWriter::try_new(&mut dest.data, arrow_schema, None)
+                .context(EncodeMemtableSnafu)?;
+            writer
+                .write(&arrow_record_batch)
+                .context(EncodeMemtableSnafu)?;
+            let _metadata = writer.finish().context(EncodeMemtableSnafu)?;
+        }
+
+        dest.metadata = BulkPartMeta {
+            num_rows: arrow_record_batch.num_rows(),
+            max_timestamp: max_ts,
+            min_timestamp: min_ts,
+        };
+        Ok(true)
+    }
+
+    /// Decodes [BulkPart] to [Batch]es.
+    fn decode_to_batches(&self, _part: &BulkPart, _dest: &mut VecDeque<Batch>) -> Result<()> {
+        todo!()
+    }
+}
+
+/// Converts mutations to record batches.
+fn mutations_to_record_batch(
+    mutations: &[Mutation],
+    metadata: &RegionMetadataRef,
+    pk_encoder: &McmpRowCodec,
+    dedup: bool,
+) -> Result<Option<(RecordBatch, i64, i64)>> {
+    let total_rows: usize = mutations
+        .iter()
+        .map(|m| m.rows.as_ref().map(|r| r.rows.len()).unwrap_or(0))
+        .sum();
+
+    if total_rows == 0 {
+        return Ok(None);
+    }
+
+    let mut pk_builder = BinaryBuilder::with_capacity(total_rows, 0);
+
+    let mut ts_vector: Box<dyn MutableVector> = metadata
+        .time_index_column()
+        .column_schema
+        .data_type
+        .create_mutable_vector(total_rows);
+    let mut sequence_vector = UInt64Builder::with_capacity(total_rows);
+    let mut op_type_vector = UInt8Builder::with_capacity(total_rows);
+
+    let mut field_builders: Vec<Box<dyn MutableVector>> = metadata
+        .field_columns()
+        .map(|f| f.column_schema.data_type.create_mutable_vector(total_rows))
+        .collect();
+
+    let mut pk_buffer = vec![];
+    for m in mutations {
+        let Some(key_values) = KeyValuesRef::new(metadata, m) else {
+            continue;
+        };
+
+        for row in key_values.iter() {
+            pk_buffer.clear();
+            pk_encoder.encode_to_vec(row.primary_keys(), &mut pk_buffer)?;
+            pk_builder.append_value(pk_buffer.as_bytes());
+            ts_vector.push_value_ref(row.timestamp());
+            sequence_vector.append_value(row.sequence());
+            op_type_vector.append_value(row.op_type() as u8);
+            for (builder, field) in field_builders.iter_mut().zip(row.fields()) {
+                builder.push_value_ref(field);
+            }
+        }
+    }
+
+    let sorter = ArraysSorter::new(
+        metadata,
+        pk_builder.finish(),
+        ts_vector.to_vector().to_arrow_array(),
+        sequence_vector.finish(),
+        op_type_vector.finish(),
+        field_builders
+            .iter_mut()
+            .map(|f| f.to_vector().to_arrow_array()),
+        dedup,
+    );
```

Comment:
We don't need to define a constructor for the `ArraysSorter`. We could fill fields of the `ArraysSorter` directly.